DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DelegateApprovalsController.cs

```diff
@@ -27,8 +27,24 @@ public IActionResult Index()
                 .GetUnapprovedDelegatesWithCustomPromptAnswersForCentre(centreId)
                 .Select(d => new UnapprovedDelegate(d.delegateUser, d.prompts));
 
-            var model = new DelegateApprovalsViewModel(delegates);
+            var model = new DelegateApprovalsViewModel(centreId, delegates);
             return View(model);
         }
+
+        [HttpPost]
+        [Route("/TrackingSystem/Delegates/Approve")]
+        public IActionResult ApproveDelegate(int delegateId)
+        {
+            delegateApprovalsService.ApproveDelegate(delegateId);
+            return RedirectToAction("Index", "DelegateApprovals");
+        }
+
+        [HttpPost]
+        [Route("/TrackingSystem/Delegates/Approve/ApproveForCentre")]
+        public IActionResult ApproveDelegatesForCentre(int centreId)
+        {
+            delegateApprovalsService.ApproveAllUnapprovedDelegatesForCentre(centreId);
+            return RedirectToAction("Index", "DelegateApprovals");
+        }

```

Comment:
I don't think the user should ever be able to approve delegates in a centre that's other than what they've logged into, so might make sense to make this a GET endpoint /TrackingSystem/Delegates/Approve/All 

In fact, you could also avoid the POST in single delegate approve by adding the ID as a query string. Then the buttons can just be links to the relevant URLs.

Reply:
I wanted to use POST since that's the type of operation being performed, but I'm not wedded to it. However, I can just get the centreId from the user claims in this endpoint either way.